### PR TITLE
fix: remove instance field from singleton IndexHtmlRequestHandler

### LINKED ISSUE #9287: IndexHtmlRequestHandler: com.vaadin.flow.server.VaadinService#modifyIndexHtmlResponse might be called for IndexHtmlResponse of other user
Method `IndexHtmlRequestHandler.synchronizedHandleRequest` stores created `indexHtmlResponse` into an instance field of this singleton-scoped class (see lines 81 and 87). It is retrieved later to call `VaadinService.modifyIndexHtmlResponse()` (line 117). However, since this is an instance field and no synchronization is performed, at the time of calling, the content of `indexHtmlResponse` might belong to another user. 

Depending on the implementation of particular `IndexHtmlRequestListener`, this may lead to sensitive data exposure or corruption (`IndexHtmlResponse` contains references of `VaadinRequest` and `VaadinResponse`, together with `index.html` document that can be modified).

https://github.com/vaadin/flow/blob/f617010cdff589bf8b8d714081d1a2d1488ca156/flow-server/src/main/java/com/vaadin/flow/server/communication/IndexHtmlRequestHandler.java#L65-L130

### PR DESCRIPTION
**Pull request by haijian-vaadin**
Fixes #9287

### CODE REVIEW

File: flow-server/src/main/java/com/vaadin/flow/server/communication/IndexHtmlRequestHandler.java [hunk 1]
```diff
@@ -59,12 +59,11 @@
  */
 public class IndexHtmlRequestHandler extends JavaScriptBootstrapHandler {
 
-    private transient IndexHtmlResponse indexHtmlResponse;
-
     @Override
     public boolean synchronizedHandleRequest(VaadinSession session,
             VaadinRequest request, VaadinResponse response) throws IOException {
         DeploymentConfiguration config = session.getConfiguration();
+        IndexHtmlResponse indexHtmlResponse = null;
```
Comment (fluorumlabs) [inline flow-server/src/main/java/com/vaadin/flow/server/communication/IndexHtmlRequestHandler.java:66]: Does it has to be initialized with `null`?

Comment (haijian-vaadin) [reply flow-server/src/main/java/com/vaadin/flow/server/communication/IndexHtmlRequestHandler.java:66]: No, but not initialized, i.e. `IndexHtmlResponse indexHtmlResponse;` is basically the same as `IndexHtmlResponse indexHtmlResponse = null;`? Or do you mean sth else?

Comment (fluorumlabs) [reply flow-server/src/main/java/com/vaadin/flow/server/communication/IndexHtmlRequestHandler.java:66]: Not really, there will be an `LDC` opcode to load variable with `null` value. Semantically, if `null` is not supposed to be there, it should be omitted, to reduce confusion during future refactoring

Comment (haijian-vaadin) [reply flow-server/src/main/java/com/vaadin/flow/server/communication/IndexHtmlRequestHandler.java]: ok, done

**Review by joheriks — APPROVED** [review]

**Pull request merged**